stripe/src/main/java/com/stripe/android/view/CountryAutoCompleteTextView.kt

```diff
@@ -58,6 +60,30 @@ internal class CountryAutoCompleteTextView @JvmOverloads constructor(
 
         selectedCountry = countryAdapter.firstItem
         updateInitialCountry()
+
+        val errorMessage = resources.getString(R.string.address_country_invalid)
+        countryAutocomplete.validator = object : AutoCompleteTextView.Validator {
+            override fun fixText(invalidText: CharSequence?): CharSequence {
+                return invalidText ?: ""
+            }
+
+            override fun isValid(text: CharSequence?): Boolean {
+                val validCountry = countryAdapter.unfilteredCountries.firstOrNull {
+                    it.name == text.toString()
+                }
+
+                return if (validCountry != null) {
+                    selectedCountry = validCountry
+                    clearError()
+                    true
+                } else {
+                    selectedCountry = null
+                    tl_country_cat.error = errorMessage
+                    tl_country_cat.isErrorEnabled = true
+                    false
```

Comment:
`selectedCountry = validCountry` can be done outside the if statement and doesn't need to be done in the else statement

and we can probably do the return outside the if statement too - `return validCountry != null`